lightning/src/ln/channelmanager.rs

```diff
@@ -1325,6 +1329,7 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 				msg: update
 			});
 		}
+		self.pending_events.lock().unwrap().push(events::Event::ChannelClosed { channel_id: *channel_id,  err: format!("") });
```

Comment:
Could multiple user calls to `force_close_channel` result in multiple `ChannelClosed` events being generated? In general, I think any situations where duplicate events could be generated is worth documenting :)

Reply:
Yes I agree with you, I think it's possible in this case if the user calls twice `force_close_channel` before getting the `CommitmentTxBroadcasted` monitor event, effectively removing the chan from our internal `channel_state.by_id`

I would eager to modify `force_close_channel` to return an error if the `short_to_id` index is empty of the chan ref but I think a user might be interested to force-close multiple times in some deployment ? Let's say when the components (channel_manager/broadcaster) are on multiple hosts and you might have a failure between them ?

Reply:
At least in the case of force-closing, it seems like we should only generate a `ChannelClosed` event if we closed the channel, which we know by whether the `channel_state.by_id` is occupied, above.

Reply:
Make the generation of the `ChannelClosed` event in `force_close_channel_with_peer` conditional on the `channel_state.by_id` slot being occupied.